notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.notifier.modules
+
+import com.atlassian.jira.rest.client.api.JiraRestClient
+import com.atlassian.jira.rest.client.api.RestClientException
+import com.atlassian.jira.rest.client.api.domain.BasicIssue
+import com.atlassian.jira.rest.client.api.domain.Comment
+import com.atlassian.jira.rest.client.api.domain.input.IssueInputBuilder
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory
+
+import java.net.URI
+
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.config.JiraConfiguration
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+
+class JiraNotifier(
+    val config: JiraConfiguration,
+    private val restClient: JiraRestClient = AsynchronousJiraRestClientFactory()
+        .createWithBasicHttpAuthentication(URI(config.host), config.username, config.password)
+) {
+    /**
+     * Returns a [ProjectIssueBuilder] object, which can be used to do operations for the given [projectKey].
+     */
+    fun issueBuilder(projectKey: String): ProjectIssueBuilder =
```

Comment:
Minor nit: Should this function now better be called `projectIssueBuilder()`? Sorry for not noticing earlier.

Reply:
Yes, it makes sense. I changed the function name to `projectIssueBuilder`.